### ISSUE TITLE
studio: shuffling slides sometimes does not work

### ISSUE THREAD

**Issue by peterpeterparker** [deckgo/deckdeckgo#1082]
### Affected part of DeckDeckGo

- [X] Studio (online editor)

### Feature Description

Changing the ordering of the slides does not work sometimes.

I tried to reproduce the issue but, couldn't.

Therefore, Iif anyone can provide an accurate description on the cases which lead to the issue, that would help a lot!

### Screenshots

<img width="1221" alt="Capture d’écran 2021-02-18 à 07 43 21" src="https://user-images.githubusercontent.com/16886711/108316338-13cea700-71bd-11eb-93f2-a21dc91427f9.png">

**Comment by zh-betina** [comment]
@peterpeterparker Hey! I'd like to have a look at that one 😉 Not sure if I'll find it, but trying never hurts 😄

**Comment by peterpeterparker** [comment]
That would  be awesome @zh-betina 👍

I did not manage to reproduce it but, I am sure the bug exists.

If you can manage to track it down, code wise, the bug might be in [updateDeckSlidesOrder](https://github.com/deckgo/deckdeckgo/blob/05a900a9161f93576a1c6f8420bca99947f57d77/studio/src/app/handlers/editor/events/deck/deck-events.handler.ts#L933)

Keep me posted and thx!

**Comment by zh-betina** [comment]
@peterpeterparker I'm not sure if I found exactly the same issue as yours, however, there is a problem with changing an order of slides, if you drag one of the elements on the list of slides too much down to the bottom. It sort of blocks the dragging event, can't drag anything anymore after and if you get off the list/click anywhere else, and go back to the list of slides, it turns out the order wasn't changed. Not sure I'm clear here but just try to take any element from the list of slides, drag it as if you wanted to move it to the last position and if you do it fast and "to the max", it won't change the order.

Haven't checked it codewise yet. Please, let me know if that could be the issue you were referring to. If it's not clear enough, I'll try to record the screen!

I used Chrome browser on Windows.

**Comment by peterpeterparker** [comment]
thanks for the analysis and feedback @zh-betina 

it might be the issue indeed, actually I don't have more information than "it doesn't work sometimes".

no need of screencast, I was able to reproduce the issue too. spontaneously I would say that it is not an issue with our code but maybe a bug of the Ionic [reorder](https://ionicframework.com/docs/api/reorder) component we are using.

do you want to have a look?

**Comment by zh-betina** [comment]
@peterpeterparker Yes! 'd like to get more familiar with the project from the code side and help with more things if my skills allow (sooner or later :)),  so I'll be happy to have a look!

**Comment by peterpeterparker** [comment]
Awesome! Go for it, ping me if you have any questions and thx 🙏

**Comment by zh-betina** [comment]
@peterpeterparker Hey, I followed the steps:

`git clone https://github.com/deckgo/deckdeckgo`
`cd deckdeckgo/studio`
`npm install`
`npm run start`

but that's the result I got on `npm start`
![printscreen](https://user-images.githubusercontent.com/62478957/113462039-5a036100-941f-11eb-93dc-dc51eafd4c74.png)

Is it something common and I missed something?

**Comment by peterpeterparker** [comment]
@zh-betina are you on Windows? another contributor, Christian, had the same issue yesterday with the docs and we came to the conclusion that was the reason

to solve the error I think it might need the `rollup-plugin-node-polyfills` as displayed in the Stencil doc https://stenciljs.com/docs/module-bundling chapter "Node Polyfills "

1. `npm install rollup-plugin-node-polyfills --save-dev`

2. add `import nodePolyfills from 'rollup-plugin-node-polyfills';` and 

```
rollupPlugins: {
    after: [
      nodePolyfills(),
    ]
  }
  ```
  
  Can you give it a try? If it work out we can add it generally to our apps.

**Comment by zh-betina** [comment]
@peterpeterparker Yes, I tried it on Windows, did as you recommended and it's working! Thanks a lot.
I'll be back soon with a more issue-related bug content (I hope 😄 ).

**Comment by peterpeterparker** [comment]
nice cool to hear it worked out. maybe at some point we should think about adding it...

looking forward to your feedback about the issue!
